pyproject.toml

```diff
@@ -27,7 +27,7 @@ s3fs = ">=2023.10.0"
 zarr = "^2.16.1"
 rioxarray = "^0.15.0"
 rasterio = "^1.3.9"
-h5py = "^3.10.0"
```

Comment:
@samland1116 was this intentional?  @samlamont I think you just updated this package, right? Any issues updating further?

Reply:
I had to pin the `netcdf4` version in PR #256 due to a weird HDF5 error related to Kerchunk.  I don't think there are issues with updating `h5py` but we should double-check

Reply:
I guess i don't have strong feelings, but if it was not intentional, maybe we just roll it back.

Reply:
Per our discussion 10/23 -- this was an intentional update to resolve a Windows-specific error with the h5py package. Though not currently developing on Windows, seems like a good bug to squash while it is fresh on our minds.